packages/react-reconciler/src/ReactFiberWorkLoop.old.js

```diff
@@ -3184,9 +3185,15 @@ function doubleInvokeEffectsInDEV(
 ) {
   const isStrictModeFiber = fiber.type === REACT_STRICT_MODE_TYPE;
   const isInStrictMode = parentIsInStrictMode || isStrictModeFiber;
+
   if (fiber.flags & PlacementDEV || fiber.tag === OffscreenComponent) {
     setCurrentDebugFiberInDEV(fiber);
-    if (isInStrictMode) {
+    const isNotOffscreen = fiber.tag !== OffscreenComponent;
+    // Checks if Offscreen is being revealed. For all other components, evaluates to true.
+    const hasOffscreenBecomeVisible =
+      isNotOffscreen ||
+      (fiber.flags & Visibility && fiber.memoizedState === null);
+    if (isInStrictMode && hasOffscreenBecomeVisible) {
```

Comment:
Factoring nit: This looks logically correct to me but the extra `isNotOffscreen` check is a bit confusing. I think you combine it with the outer check like this:

```js
if (
  isInStrictMode &&
  // Check for newly mounted trees
  (fiber.flags & PlacementDEV ||
    // Check for trees that are going from hidden -> visible
    (fiber.tag === OffscreenComponent &&
      fiber.flags & Visibility &&
      fiber.memoizedState === null))
) {
  // Double invoke all the effects in this tree
  setCurrentDebugFiberInDEV(fiber);
  disappearLayoutEffects(fiber);
  disconnectPassiveEffect(fiber);
  reappearLayoutEffects(root, fiber.alternate, fiber, false);
  reconnectPassiveEffects(root, fiber, NoLanes, null, false);
  resetCurrentDebugFiberInDEV();
} else {
  // Continue searching for newly mounted or newly revealed trees
  recursivelyTraverseAndDoubleInvokeEffectsInDEV(root, fiber, isInStrictMode);
}
```

Reply:
Nvm I guess that doesn't check if Offscreen is hidden when it has a PlacementDEV flag set. Just merge as-is and I can submit a separate PR if I have a suggestion.

Reply:
I tried simplifying this. I think the complexity with extra variables is not terrible. It is fairly well covered with unit tests.